### ISSUE TITLE
[Research] LoRA, QLoRA & base model selection for translation

### ISSUE THREAD

**Issue by selcuk-yalcin** [selcuk-yalcin/Kurdish-LLM-FineTuning#4]
## Summary

Research and document the fine-tuning approach for **Stage A (Translation)**: which base LLM to use, **LoRA vs QLoRA**, hyperparameters, and VRAM requirements. Output feeds directly into `config/translation/` before Vast.ai training.

## Current default

```yaml
# config/translation/base.yaml
model:
  name: meta-llama/Llama-3.2-1B-Instruct
quantization:
  load_in_4bit: false   # set true for QLoRA on 7B models
```

### 1. LoRA vs QLoRA

| Method | VRAM | Quality | When to use |
|--------|------|---------|-------------|
| **LoRA** (16-bit) | Higher | Best for small models | 1B–2B on 16–24 GB GPU |
| **QLoRA** (4-bit base + LoRA) | Lower | Slightly lower | 7B+ on 24 GB GPU |

- Read PEFT LoRA guide and QLoRA paper
- Decide: stay on **Llama-3.2-1B** for fast iteration, or upgrade to **Qwen2.5-1.5B** / **Mistral-7B QLoRA** for quality

### 2. Base model candidates

| Model | Params | Notes |
|-------|--------|-------|
| `meta-llama/Llama-3.2-1B-Instruct` | 1B | Current choice — fast, limited Kurdish |
| `Qwen/Qwen2.5-1.5B-Instruct` | 1.5B | Strong multilingual |
| `google/gemma-2-2b-it` | 2B | Good instruction tuning |
| `mistralai/Mistral-7B-Instruct-v0.3` | 7B | Best quality, needs QLoRA |

### 3. Training stack (already in repo)

- **PEFT** — LoRA adapters
- **TRL SFTTrainer** — supervised fine-tuning
- **bitsandbytes** — 4-bit quantization (CUDA only, skipped on macOS)

## Tasks

- [ ] Read resources below (LoRA + QLoRA)
- [ ] Run a **local smoke test** on Mac MPS if possible: 10–50 steps with current config
- [ ] Document decision in `config/translation/base.yaml` + short note in issue
- [ ] Finalize `config/translation/lora.yaml` (rank, alpha, target modules)
- [ ] Finalize `config/translation/training.yaml` (epochs, batch size, learning rate)
- [ ] Share summary with @cem-acun for review

## Acceptance criteria

- Written decision: model + LoRA/QLoRA + expected VRAM
- Config files updated and committed
- Ready for Phase 2 Vast.ai full training run

## Parallel work

**Independent:** Can run in parallel with @cem-acun’s Vast.ai setup ([#2](https://github.com/selcuk-yalcin/Kurdish-LLM-FineTuning/issues/2)) and data spot-check ([#3](https://github.com/selcuk-yalcin/Kurdish-LLM-FineTuning/issues/3)).

### Papers
- [LoRA: Low-Rank Adaptation of Large Language Models (Hu et al., 2021)](https://arxiv.org/abs/2106.09685)
- [QLoRA: Efficient Finetuning of Quantized LLMs (Dettmers et al., 2023)](https://arxiv.org/abs/2305.14314)

### Docs
- [Hugging Face PEFT — LoRA conceptual guide](https://huggingface.co/docs/peft/main/en/conceptual_guides/lora)
- [Hugging Face TRL — SFT Trainer](https://huggingface.co/docs/trl/main/en/sft_trainer)
- [bitsandbytes — 4-bit quantization](https://huggingface.co/docs/bitsandbytes/main/en/index)

### Models
- [Llama 3.2 1B Instruct](https://huggingface.co/meta-llama/Llama-3.2-1B-Instruct)
- [Qwen2.5-1.5B-Instruct](https://huggingface.co/Qwen/Qwen2.5-1.5B-Instruct)
- [Mistral-7B-Instruct-v0.3](https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.3)

## Assignee

@selcuk-yalcin — ML config and smoke tests  
@cem-acun — optional: review model choice from Kurdish language perspective

**Comment by selcuk-yalcin** [comment]
Consolidated into streamlined 35-issue roadmap (Aug 2026).
